crates/wasmparser/src/validator.rs

```diff
@@ -410,8 +410,9 @@ impl Validator {
             section,
             "type",
             |state, _, count, offset| {
+                state.module.assert_mut().types.0.reserve(count as usize);
```

Comment:
I think this reservation will want to come after `check_max` because otherwise if `count` is huge I think it could cause OOMs?

Reply:
I'll fix that.